tests/periph_gpio_arduino/main.sketch

```diff
@@ -0,0 +1,197 @@
+        printf("usage: %s <pin>\n", argv[0]);
+        return 1;
+    }
+
+    if (get_digital_pin(argv[1], &pin) != 0) {
+        printf("Could not map Arduino pin %s to a gpio.\nExpected numeric pin number.\n", argv[1]);
+        return 1;
+    }
+
+    pinMode(pin, mode);
+
+    return 0;
+}
+
+static int init_out(int argc, char **argv)
+{
+    return init_pin(argc, argv, OUTPUT);
+}
+
+static int init_in(int argc, char **argv)
+{
+    return init_pin(argc, argv, INPUT);
+}
+
+static int init_in_pu(int argc, char **argv)
+{
+    return init_pin(argc, argv, INPUT_PULLUP);
+}
+
+static int read(int argc, char **argv)
+{
+    int pin;
+
+    if (argc < 2) {
+        printf("usage: %s <pin>\n", argv[0]);
+        return 1;
+    }
+
+    if (get_digital_pin(argv[1], &pin) != 0) {
+        printf("Could not map Arduino pin %s to a gpio.\nExpected numeric pin number.\n", argv[1]);
+        return 1;
+    }
+
+    if (digitalRead(pin)) {
+        printf("ARDUINO_PIN(%i) is HIGH\n", pin);
+    }
+    else {
+        printf("ARDUINO_PIN(%i) is LOW\n", pin);
+    }
+
+    return 0;
+}
+
+static int set(int argc, char **argv)
+{
+    int pin;
+
+    if (argc < 2) {
+        printf("usage: %s <pin>\n", argv[0]);
+        return 1;
+    }
+
+    if (get_digital_pin(argv[1], &pin) != 0) {
+        printf("Could not map Arduino pin %s to a gpio.\nExpected numeric pin number.\n", argv[1]);
+        return 1;
+    }
+
+
+    digitalWrite(pin, 1);
+
+    return 0;
+}
+
+static int clear(int argc, char **argv)
+{
+    int pin;
+
+    if (argc < 2) {
+        printf("usage: %s <pin>\n", argv[0]);
+        return 1;
+    }
+
+    if (get_digital_pin(argv[1], &pin) != 0) {
+        printf("Could not map Arduino pin %s to a gpio.\nExpected numeric pin number.\n", argv[1]);
+        return 1;
+    }
+
+
+    digitalWrite(pin, 0);
+
+    return 0;
+}
+
+static int toggle(int argc, char **argv)
+{
+    int pin;
+
+    if (argc < 2) {
+        printf("usage: %s <pin>\n", argv[0]);
+        return 1;
+    }
+
+    if (get_digital_pin(argv[1], &pin) != 0) {
+        printf("Could not map Arduino pin %s to a gpio.\nExpected numeric pin number.\n", argv[1]);
+        return 1;
+    }
+
+    // not thread safe but Arduino doesn't promise such things
+    if (digitalRead(pin)) {
+        digitalWrite(pin, 0);
+    }
+    else {
+        digitalWrite(pin, 1);
+    }
+
+    return 0;
+}
+
+static const shell_command_t shell_commands[] = {
+    { "init_out", "init as output (push-pull mode)", init_out },
+    { "init_in", "init as input w/o pull resistor", init_in },
+    { "init_in_pu", "init as input with pull-up", init_in_pu },
+    { "read", "read pin status", read },
+    { "set", "set pin to HIGH", set },
+    { "clear", "set pin to LOW", clear },
+    { "toggle", "toggle pin", toggle },
+    { NULL, NULL, NULL }
+};
+
+void setup(void)
+{
+    puts("Arduino GPIO peripheral driver test\n"
+         "In this test, pins are specified by integer pin numbers.\n"
```

Comment:
I'd rather add a proper README.md for the test instead of putting the description in the binary, the latter also increases the binary size which might be problematic considering small nodes such as Arduino-Uno